cocos2d/core/sprites/CCSprite.js

```diff
@@ -907,33 +895,28 @@ cc.Sprite = cc.Node.extend(/** @lends cc.Sprite# */{
         if(isFileName)
             texture = cc.textureCache.addImage(texture);
 
-        if(texture._textureLoaded){
-            this._setTexture(texture, isFileName);
-            this.setColor(this._realColor);
-            this._textureLoaded = true;
-        }else{
-            this._renderCmd._setTexture(null);
-            texture.addEventListener("load", function(){
-                this.setNodeDirty(true);
-                this._setTexture(texture, isFileName);
-                this.setColor(this._realColor);
-                this._textureLoaded = true;
+        this._loader.clear();
+        if (!texture._textureLoaded) {
+            // wait for the load to be set again
+            this._loader.add(texture, function () {
+                this.setTexture(texture);
+                this.dispatchEvent("load");
```

Comment:
This should be dispatched at the end of this function, after `this._textureLoaded = true;`